simulation/model.py

```diff
@@ -59,13 +64,19 @@ def heating_systems(self) -> Set[HeatingSystem]:
     @property
     def air_source_heat_pump_discount_factor(self) -> float:
 
-        if self.current_datetime.year < 2022:
-            return 1
-        if self.current_datetime.year > 2022:
-            return 1 - self.air_source_heat_pump_discount_factor_2022
-        else:
-            month = self.current_datetime.month
-            return 1 - (month / 12 * self.air_source_heat_pump_discount_factor_2022)
+        dates = list(self.air_source_heat_pump_price_discount_schedule.keys())
+        factors = list(self.air_source_heat_pump_price_discount_schedule.values())
+        date_ranges = list(zip(dates, dates[1:]))
+
+        for idx, date_range in enumerate(date_ranges):
+            if date_range[idx] <= self.current_datetime <= date_range[idx + 1]:
+                proportion_through_range = (self.current_datetime - date_range[0]) / (
+                    date_range[1] - date_range[0]
+                )
+                return (
+                    factors[idx]
+                    + (factors[idx + 1] - factors[idx]) * proportion_through_range
+                )
```

Comment:
Just a note that I'm not keen on this but need to have a think...

Reply:
I agree we should use `bisect`. I think its up to us to decide if we want to linearly interpolate the costs between the dates or just have hard filters where it becomes active at a certain date onwards. I can see the pros and cons of both modelling choices.